commons/src/tracing.rs

```diff
@@ -0,0 +1,113 @@
+//! Tracing service.
+
+use opentelemetry::api::{
+    Carrier, HttpTextFormat, Key, Provider, Span, SpanContext, TraceContextPropagator,
+};
+use opentelemetry::{global, sdk};
+use opentelemetry_jaeger::{Exporter, Process};
+use std::collections::HashMap;
+
+use actix_web::dev::ServiceRequest;
+use actix_web::http;
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+
+use crate::prelude_errors::*;
+
+/// init_tracer sets up Jaeger tracer
+pub fn init_tracer(name: &'static str, maybe_agent_endpoint: Option<String>) -> Fallible<()> {
+    // Skip provider config if agent endpoint is not set
+    let agent_endpoint = match maybe_agent_endpoint {
+        None => return Ok(()),
+        Some(s) => s,
+    };
+
+    let exporter = match Exporter::builder()
+        .with_agent_endpoint(agent_endpoint)
+        .with_process(Process {
+            service_name: name.to_string(),
+            tags: vec![Key::new("exporter").string("jaeger")],
+        })
+        .init()
+    {
+        Ok(e) => e,
+        Err(e) => return Err(e.into()),
+    };
```

Comment:
Did the following not work here?

```suggestion
        .init()?:
```

Reply:
It did, but it was throwing `thrift::error`. It probably should stay as an implementation detail of `init_tracer`, thus this commit with a `match`

Reply:
Weird, I just tried it locally on top of your branch with the following diff:

```diff
diff --git a/commons/src/tracing.rs b/commons/src/tracing.rs
index 5f0c4d9a..9b7c1ba7 100644
--- a/commons/src/tracing.rs
+++ b/commons/src/tracing.rs
@@ -21,17 +21,13 @@ pub fn init_tracer(name: &'static str, maybe_agent_endpoint: Option<String>) ->
         Some(s) => s,
     };

-    let exporter = match Exporter::builder()
+    let exporter = Exporter::builder()
         .with_agent_endpoint(agent_endpoint)
         .with_process(Process {
             service_name: name.to_string(),
             tags: vec![Key::new("exporter").string("jaeger")],
         })
-        .init()
-    {
-        Ok(e) => e,
-        Err(e) => return Err(e.into()),
-    };
+        .init()?;
```

Reply:
See my latest review, it works for me like that.

Reply:
Oh, got it, Rust can autoconvert this to `Fallible`, pushed bf73b0c576e4acd4ac2fbfc8af5ed9864d807e8e